contracts/src/warpDiscordBot/messagesContent/write/addMessage.ts

```diff
@@ -0,0 +1,94 @@
+import { validateInputArgumentPresence, validateString } from '../../../utils';
+import { ContractAction, ContractState, ContractResult, messagesPrefix, counterPrefix } from '../../types/types';
+
+declare const ContractError;
+declare const SmartWeave;
+
+export const addMessage = async (
+  state: ContractState,
+  { input: { id, messageId, content, roles } }: ContractAction
+): Promise<ContractResult> => {
+  validateInputArgumentPresence(id, 'id');
+  validateString(id, 'id');
+  validateInputArgumentPresence(messageId, 'messageId');
+  validateString(messageId, 'messageId');
+  validateInputArgumentPresence(content, 'content');
+  validateInputArgumentPresence(roles, 'roles');
+
+  let effectiveContent: string = '';
+  if (content.length > 2000) {
+    effectiveContent = content.substring(0, 1999);
+  } else {
+    effectiveContent = content;
+  }
+
+  const effectiveCaller = `${id}_${messageId}_${SmartWeave.block.timestamp}`;
```

Comment:
as discussed - I'm leaving it as it is right now as it would be quite difficult to filter out message which needs to be removed when running `removeMessage` interaction.